### ISSUE TITLE
Health must be between 0 and 20.0, but was 99.0

### ISSUE THREAD

**Issue by s1h4d0w** [Nathat23/StackMob-5#190]
I've been getting this error randomly in console, not sure if it's a StackMob error or an error generated when a player kills an [Infernal Mob](https://www.spigotmc.org/resources/infernal-mobs.2156/). That plugin transforms some normally spawning mobs into a much more powerful boss mob, I could see that being the culprit.

Running StackMob 5.4.2 on Paper 472 1.16.5. Just saw there's a 5.4.2 version but didn't see anything in the changelog that seems like it would fix this, might be wrong.

```
[01:19:30 WARN]: [StackMob] Task #99710 for StackMob v5.4.2 generated an exception
java.lang.IllegalArgumentException: Health must be between 0 and 30.0, but was 148.39999389648438. (attribute base value: 30.0)
at org.bukkit.craftbukkit.v1_16_R3.entity.CraftLivingEntity.setHealth(CraftLivingEntity.java:117) ~[patched_1.16.5.jar:git-Paper-472]
at uk.antiperson.stackmob.entity.death.KillStepDamage.onSpawn(KillStepDamage.java:43) ~[?:?]
at uk.antiperson.stackmob.listeners.DeathListener.lambda$onStackDeath$0(DeathListener.java:64) ~[?:?]
at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:100) ~[patched_1.16.5.jar:git-Paper-472]
at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-472]
at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1294) ~[patched_1.16.5.jar:git-Paper-472]
at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:377) ~[patched_1.16.5.jar:git-Paper-472]
at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1209) ~[patched_1.16.5.jar:git-Paper-472]
at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:997) ~[patched_1.16.5.jar:git-Paper-472]
at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:174) ~[patched_1.16.5.jar:git-Paper-472]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]
```

**Comment by s1h4d0w** [comment]
More of the same errors:

```
[01:21:00 WARN]: [StackMob] Task #1506737 for StackMob v5.4.2 generated an exception
java.lang.IllegalArgumentException: Health must be between 0 and 30.0, but was 146.0. (attribute base value: 30.0)
        at org.bukkit.craftbukkit.v1_16_R3.entity.CraftLivingEntity.setHealth(CraftLivingEntity.java:117) ~[patched_1.16.5.jar:git-Paper-509]
        at uk.antiperson.stackmob.entity.death.KillStepDamage.onSpawn(KillStepDamage.java:43) ~[?:?]
        at uk.antiperson.stackmob.listeners.DeathListener.lambda$onStackDeath$0(DeathListener.java:64) ~[?:?]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:99) ~[patched_1.16.5.jar:git-Paper-509]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1299) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:377) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1214) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1002) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:175) ~[patched_1.16.5.jar:git-Paper-509]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]
[01:22:33 INFO]: Ronaldo423 issued server command: /ec
[01:22:59 INFO]: Ronaldo423 issued server command: /top
[01:23:02 INFO]: Ronaldo423 issued server command: /ec
[01:24:16 INFO]: Ronaldo423 issued server command: /top
[01:24:16 INFO]: &d[NOTE] &fCheck &awww.lepelcraft.nl &ffor information on commands! 
[01:24:44 INFO]: Ronaldo423 issued server command: /ec
[01:27:19 INFO]: Ronaldo423 issued server command: /ec
[01:27:24 INFO]: Ronaldo423 issued server command: /shop
[01:28:34 INFO]: Ronaldo423's Robinhood's Bow killed an Guardian (10)!
[01:28:34 WARN]: [StackMob] Task #1536905 for StackMob v5.4.2 generated an exception
java.lang.IllegalArgumentException: Health must be between 0 and 30.0, but was 141.0. (attribute base value: 30.0)
        at org.bukkit.craftbukkit.v1_16_R3.entity.CraftLivingEntity.setHealth(CraftLivingEntity.java:117) ~[patched_1.16.5.jar:git-Paper-509]
        at uk.antiperson.stackmob.entity.death.KillStepDamage.onSpawn(KillStepDamage.java:43) ~[?:?]
        at uk.antiperson.stackmob.listeners.DeathListener.lambda$onStackDeath$0(DeathListener.java:64) ~[?:?]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:99) ~[patched_1.16.5.jar:git-Paper-509]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1299) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:377) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1214) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1002) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:175) ~[patched_1.16.5.jar:git-Paper-509]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]
[01:29:13 INFO]: Ronaldo423 issued server command: /ec
[01:32:53 INFO]: Ronaldo423 issued server command: /feed
[01:34:16 INFO]: &d[NOTE] &fVote for lepelcraft and earn &a$1500 per vote&f or even a &egenie lamp&f! Press T and click on: &ahttps://www.lepelcraft.nl/vote-1-14-4/ Make sure you are online to receive rewards!
[01:37:03 WARN]: [StackMob] Task #1567210 for StackMob v5.4.2 generated an exception
java.lang.IllegalArgumentException: Health must be between 0 and 30.0, but was 146.0. (attribute base value: 30.0)
        at org.bukkit.craftbukkit.v1_16_R3.entity.CraftLivingEntity.setHealth(CraftLivingEntity.java:117) ~[patched_1.16.5.jar:git-Paper-509]
        at uk.antiperson.stackmob.entity.death.KillStepDamage.onSpawn(KillStepDamage.java:43) ~[?:?]
        at uk.antiperson.stackmob.listeners.DeathListener.lambda$onStackDeath$0(DeathListener.java:64) ~[?:?]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:99) ~[patched_1.16.5.jar:git-Paper-509]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1299) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:377) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1214) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1002) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:175) ~[patched_1.16.5.jar:git-Paper-509]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]
[01:37:08 WARN]: [StackMob] Task #1567459 for StackMob v5.4.2 generated an exception
java.lang.IllegalArgumentException: Health must be between 0 and 30.0, but was 149.39999389648438. (attribute base value: 30.0)
        at org.bukkit.craftbukkit.v1_16_R3.entity.CraftLivingEntity.setHealth(CraftLivingEntity.java:117) ~[patched_1.16.5.jar:git-Paper-509]
        at uk.antiperson.stackmob.entity.death.KillStepDamage.onSpawn(KillStepDamage.java:43) ~[?:?]
        at uk.antiperson.stackmob.listeners.DeathListener.lambda$onStackDeath$0(DeathListener.java:64) ~[?:?]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:99) ~[patched_1.16.5.jar:git-Paper-509]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1299) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:377) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1214) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1002) ~[patched_1.16.5.jar:git-Paper-509]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:175) ~[patched_1.16.5.jar:git-Paper-509]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]
```

**Comment by TerdyTheTerd** [comment]
I have also started noting these errors on my server, as far as I am aware I did not update or change anything related to StackMob
`Health must be between 0 and 1.0, but was 13.625.`

**Comment by Nathat23** [comment]
Fixed by 56040fad46c41fe1d51c4733a915e784d2a7da98
